mlrun/projects/pipelines.py

```diff
@@ -184,6 +184,15 @@ def __init__(self):
         self.workflow_artifact_path = None
         self.runs_map = {}
 
+    def is_run_local(self, local=None):
+        force_run_local = mlrun.mlconf.force_run_local
+        if force_run_local is None or force_run_local == "auto":
+            force_run_local = False if mlrun.mlconf.namespace else True
```

Comment:
Add a method on the config called `_is_api_running_on_k8s()` and explain there your heuristic